File: android/src/main/kotlin/com/bond/whiteboard/teb/BoardAwareInterface.kt
package com.bond.whiteboard.teb

import com.tencent.teduboard.TEduBoardController.TEduBoardFileInfo

interface BoardAwareInterface {
    fun onTEBHistroyDataSyncCompleted()
    fun addBoardView()
    fun removeBoardView()
    fun setCanUndo(canUndo: Boolean)
    fun setCanRedo(canredo: Boolean)
    fun addFile(fileId: String?): TEduBoardFileInfo?
    fun onTextComponentStatusChange(id: String?, status: String?)
    fun onTEBSyncData(data: String)
    fun sendMessage(data: ByteArray, extension: String)
}

File: ios/Classes/teb/BoardAwareInterface.swift
//
// Created by vicky Leu on 2021/5/19.
//

import Foundation
protocol BoardAwareInterface {
    func onTEBHistroyDataSyncCompleted()
    func addBoardView()
    func removeBoardView()
    func setCanUndo(_ canUndo: Bool)
    func setCanRedo(_ canredo: Bool)
    func addFile(_ fileId: String?) -> TEduBoardFileInfo?
    func onTextComponentStatusChange(_ id: String?, _ status: String?)

}
